### DISCUSSION TITLE
G7 support for newer watches as standalone collector?

### DISCUSSION THREAD

**Discussion by Ihavediabeetus** [NightscoutFoundation/xDrip#3449] [General]
I recently got moved to the g7 off the g6 and can no longer use my bluejay gts at work. My employer absolutely does not allow a phone, anything with a camera, or anthing that can be plugged into a computer and have stuff saved to it past a certian point. They even check watches for cellular service before entry is allowed even though there is no signal to be had there. 

So I no longer have a good way to keep track of my bg when I am at work. The dexcom reciever will work but I know that my dumba** will eventually eave it somewhere (probably inside a tool)and never find it again as it isn't attached to me like a watch is. Also, the alarms on that thing are absolutely atrocious.

I recently picked up a Samsung watch 6(bt/wifi only)hoping that it would work with the wear support in x drip and it does perfectly.......until I change settings to make it the collector then I don't get anything until I turn off "force wear collection service". 

Because of this, I am assuming that the newer watches/watch software (wear 3.0 and up specifically) doesn't have support or a happy way around the security so xdrip can run them as a collector yet. I was just curious if newer watch support is even possible and/or a thing in the pipeline that we can hope for?

I am definitely not a programmer so I would probably end up writing code that would set a phone and watch on fire violently if i tried, probably something similar to the note hand grenade from a several years ago, lol. I am best off as just moral support and a willing guinea pig to test whatever is made whenever it is available.

Thanks.

**Comment by rpm5099** [comment]
I'm having trouble finding any information as to the availability [or existence] of a watch capable of being a standalone collector with G7.  @Navid200 - I know you are extremely busy addressing issues, if you have any insight into this it would be greatly appreciated.  I would be happy to add to the docs whatever whatever the results are, and possibly submit a pull to enable functionality (if it's within the realm of my current Android capabilities).

**Reply by Navid200** [reply]
As far as I know WearOS cannot directly collect from G7 or even Dexcom One.

There is a new BlueJay watch that is in beta test right now.  It can collect directly from G7.  https://bluejay.website/shop/product/bluejay-u1-27  
However, it is effectively a phone.  It has storage and Sim and camera ...

According to the first post, that is not acceptable.

An employer can impose reasonable restrictions.  So, an employer not hiring a diabetic as a pilot does not bother me.
But, an employer not allowing one to use a medical device while doing their job is not reasonable to me.

A possible solution for this case may be Apple watch.
https://youtu.be/7e_HbA2Vh7w?si=pRxP2l9_hTEAebY2
  
You will need an iPhone.  But, you can leave the phone behind.
When back home, you can switch to xDrip on an Android.

**Reply by heygar** [reply]
Dexcom has hinted at "direct to watch" capabilities. Currently, it's only available for Apple Watch and I am strongly opposed to joining the Apple Collective. 
As a side note, the WearOs development seems to be kind of "broken". The WearOs complication doesn't even work with the latest xdrip code. 
I was kind of hoping with Dexcom mentioning direct to watch that the G7 may overcome some of the past struggles with direct to WearOs. 

Another bit of information, I purchased a Samsung Galaxy 6 and it would occasionally (like every 20 minutes or so) get a reading directly from my G6. But, that was with a much older version of xdrip that did not support G7. Once I upgraded to G7 and grabbed the necessary version of Xdrip, it doesn't appear to work at all. :(

**Comment by osa1** [comment]
What features does a watch need to have to be able to add Dexcom 7 receiver capabilities to it?

I'm using a Garmin watch and I have [my own watch face][1] that fetches by BG levels from an online Dexcom service.

However it only works when the phone is nearby as fetching the data from the internet is done via the Garmin phone app.

I want the watch to talk directly with the sensor, without a phone.

I searched for this a little bit, and it seems like the watch needs to act as a BLE client. I found this in Garmin SDK: https://developer.garmin.com/connect-iq/api-docs/Toybox/BluetoothLowEnergy.html, it says

> The BluetoothLowEnergy module provides access to Generic BLE communication functionality in the central role

Not sure what "central role" means. Does that mean it can act as a receiver?

Any other tips on this would be appreciated.

[1]: https://github.com/osa1/Dextrack

**Reply by Navid200** [reply]
It's a challenge to support different phones as collectors.  There are some that are useless and horrible collectors.
Pixel is the best phone as a collector from Dexcom in my personal opinion.

Taking that concept to watches is not going to make it a cleaner concept.
There are no watches in production that can directly collect from a G7 other than Apple watch.
BlueJay U1 is a candidate that is very promising.  But, it is still in beta test: https://bluejay.website/shop/product/bluejay-u2-28
I just noticed that U2 is out.  I'm not even sure what it offers over U1.  
Anyway, the point is that even if you have something that theoretically works, it could be far from a reliable device the same as phones.

**Reply by osa1** [reply]
Thanks for the answer.

> There are no watches in production that can directly collect from a G7 other than Apple watch
> ...
> Anyway, the point is that even if you have something that theoretically works, it could be far from a reliable device the same as phones.

Could you please elaborate on this? Assuming the watch supports the same protocols at the hardware level (BLE or whatever Dexcom G7 is using), why would a watch not be as reliable as a phone for collecting data?

I really want to remove the phone from the equation, and I don't want Apple produccts and definitely not BlueJay (which looks like a toy compared to Apple or Garmin watches).

**Reply by Navid200** [reply]
That's not what I said.

I may get a warning for this:
There are phones that re not the best to use as an xDrip collector from a Dexcom device.  Xiaomi is an example.  
The only difference between a Xiaomi phone and a Pixel phone is what the manufacturer has done (or hasn't done) to Android on them.

The same way, all watches are not going to be the same.  Developers have been working on BlueJay for quite a while and it is still being tested.  When it is released, for the same reason, I will not buy a Xiaomi as my next phone, I will not buy any watch other than BlueJay.

**Reply by osa1** [reply]
Makes sense.

Still, does Garmin watches that support the API I linked (https://developer.garmin.com/connect-iq/api-docs/Toybox/BluetoothLowEnergy.html) can in principle work as a collector?

If they can then I would like to give this a chance and see how it works.

> That's not what I said.

Not sure what you are referring to by "that". I literally copied your message in the quoted parts.

**Reply by Navid200** [reply]
![Screenshot 2024-09-19 131730](https://github.com/user-attachments/assets/9d7a52d0-f730-4311-87cd-f37a19347f1e)

When I said I didn't say what I meant I did not make a comparison between a watch and a phone.

**Reply by osa1** [reply]
I was referring to this part:

>  Anyway, the point is that even if you have something that theoretically works, it could be far from a reliable device the same as phones.

**Reply by Navid200** [reply]
Right.

I'm just sharing some thoughts.  I am sorry I cannot give you any black and white answer.  

Some phone manufacturers do everything they can to preserve battery and in the process, kill xDrip.
Watches are worse because their batteries are much smaller.  So, it is understandable if watch manufacturers put in limitations that negatively affect xDrip.

I used Motorola phones for years.  I used xDrip on them.  Over the years, my experience had led me to conclude that the Bluetooth watchdog kicking in occasionally was just part of life.
I would be driving and the GPS on my phone would stop working because the xDrip Bluetooth watch dog had decided to turn Bluetooth off and on!
It was my belief that this was what everyone had to put up with.
Until I bought my first Pixel phone.  I have been using it for slightly over a year now.   Bluetooth watchdog?  What is that?!  It has not kicked in since.  Connectivity is unbelievably solid.  
It is not easy to figure out what the cause is.  Until you get a solid performer, you may think that average is the best you can get.  They all run the same xDrip code.

Now that I have experienced Pixel, I will never go near Samsung or Xiaomi.  

I am only thinking that this issue of intermittent connectivity due to device manufacturer trying to preserve the battery is only going to be worse for watches.  Theoretically speaking, Bluetooth exists on Pixel and Samsung and Xiaomi and they are all capable.
But, I know that is just theory.  Reality is that there is a lot the manufacturer can do that you have to jump through hoops and may still not find a solution.

**Comment by Napsterbater** [comment]
Is it still the case that a pixel watch for example cannot be a collector for a G7?

**Reply by heygar** [reply]
I switched form xdrip to Juggluco about a year ago so take this with a grain of salt. 
I couldn't get G7 to do direct to watch at all using xdrip. I assume it's still not supported/functional. At one point the xdrip app on WearOS was completely broken, but I think that's been fixed.
If "direct to watch" is important and you don't want to join the Apple Collective, I'd switch to Juggluco. It works nearly flawlessly with my Galaxy Watch 6. Only complaint is battery usage.
